grakn-core/src/main/java/ai/grakn/graql/QueryBuilder.java

```diff
@@ -104,6 +110,41 @@
     @CheckReturnValue
     <T extends Query<?>> T parse(String queryString);
 
+    /**
+     * @param queryString a string representing several queries
+     * @return a list of queries
+     */
+    @CheckReturnValue
+    <T extends Query<?>> Stream<T> parseList(String queryString);
+
+    /**
+     * @param reader a {@link Reader} representing several queries
+     * @return a list of queries
+     */
+    @CheckReturnValue
+    <T extends Query<?>> Stream<T> parseList(Reader reader);
+
+    /**
+     * @param template a string representing a templated graql query
+     * @param data data to use in template
+     * @return a query, the type will depend on the type of template.
+     */
+    @CheckReturnValue
+    <T extends Query<?>> Stream<T> parseTemplate(String template, Map<String, Object> data);
+
+    /**
+     * Register an aggregate that can be used when parsing a Graql query
+     * @param name the name of the aggregate
+     * @param aggregateMethod a function that will produce an aggregate when passed a list of arguments
+     */
+    void registerAggregate(String name, Function<List<Object>, Aggregate> aggregateMethod);
+
+    /**
+     * Register a macro that can be used when parsing a Graql template
+     * @param macro the macro to register
+     */
+    void registerMacro(Macro macro);
+
```

Comment:
These methods don't need to be here. They're still accessible with `parser().registerMacro()` or similar.

I'll put a 🔪 where things have been brought back to life that need to die.